mos_tests/neutron/python_tests/test_floating_ip.py

```diff
@@ -0,0 +1,139 @@
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import pytest
+
+from paramiko.ssh_exception import SSHException, NoValidConnectionsError
+
+from mos_tests.neutron.python_tests.base import TestBase
+
+
+@pytest.mark.usefixtures("setup")
+class TestFloatingIP(TestBase):
+    """Check association and disassociation floating ip"""
+
+    @pytest.fixture(autouse=True)
+    def _prepare_openstack(self, init):
+        """Prepare OpenStack for scenarios run
+
+        Steps:
+            1. Create network net01, subnet net01__subnet with CIDR 10.1.1.0/24
+            2. Create new security group sec_group1
+            3. Add Ingress rule for TCP protocol to sec_group1
+            4. Boot vm1 net01 with sec_group1
+        """
+        # init variables
+        exist_networks = self.os_conn.list_networks()['networks']
+        ext_net = [x for x in exist_networks
+                   if x.get('router:external')][0]
+        zone = self.os_conn.nova.availability_zones.find(zoneName="nova")
+        security_group = self.os_conn.create_sec_group_for_ssh()
+        hostname = zone.hosts.keys()[0]
+        cidr = "10.1.1.0/24"
+        self.instance_keypair = self.os_conn.create_key(key_name='instancekey')
+
+        self.os_conn.nova.security_group_rules.create(
+            security_group.id,
+            ip_protocol='tcp',
+            from_port=1,
+            to_port=65535,
+            cidr='0.0.0.0/0')
+
+        net, subnet = self.create_internal_network_with_subnet(cidr=cidr)
+        # create router
+        self.create_router_between_nets(ext_net, subnet)
+
+        self.os_conn.create_server(
+            name='server01',
+            availability_zone='{}:{}'.format(zone.zoneName,
+                                             hostname),
+            key_name=self.instance_keypair.name,
+            nics=[{'net-id': net['network']['id']}],
+            security_groups=[security_group.id])
+
+        # add floating ip to first server
+        server = self.os_conn.nova.servers.find(name="server01")
+        self.floating_ip = self.os_conn.assign_floating_ip(server,
+                                                           use_neutron=True)
+
+        pkeys = self.convert_private_key_for_vm(
+            [self.instance_keypair.private_key])
+
+        self.check_vm_is_accessible_with_ssh(
+            vm_ip=self.floating_ip['floating_ip_address'],
+            pkeys=pkeys)
+
+    def check_vm_inaccessible_by_ssh(self, vm_ip, pkeys):
+        """Check that instance is inaccessible with ssh via floating_ip.
+
+        :param vm_ip: floating_ip of instance
+        :param pkeys: ip of instance to ping
+        """
+        with pytest.raises(NoValidConnectionsError) as exc:
+            with self.env.get_ssh_to_cirros(vm_ip, pkeys) as vm_remote:
+                vm_remote.execute("date")
+        assert "Unable to connect to port 22 on  or {}".format(vm_ip) \
```

Comment:
We decided that there is no sense in this additional check of message.
So it was deleted.
Done